File: 35. Search insert position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        for i in range(0, len(nums)):
            if nums[i] == -1 and target == 0:
                return i + 1

            if nums[i] == target:
                return i
            
            if nums[i] > target:
                return i

            if nums[i - 1] < target < nums[i]:
                return i
                     
        if target > nums[len(nums) - 1]:
            return len(nums) 
        

if __name__ == "__main__":
    sol = Solution()
    nums = [-1,3,5,6]
    print("Original List : ", nums)
    
    target = 0
    index = sol.searchInsert(nums, target)
    print(index)
    

File: 35. Search_Insert_Position.js
let searchInsert = function (nums, target) {
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === -1 && target === 0) {
      return i + 1;
    }

    if (nums[i] === target) {
      return i;
    }

    if (nums[i] > target) {
      return i;
    }

    if (i > 0 && nums[i - 1] < target && target < nums[i]) {
      return i;
    }
  }

  if (target > nums[nums.length - 1]) {
    return nums.length;
  }
};

const nums = [-1, 3, 5, 6];

console.log("Original List:", nums);

const target = 0;

const result = searchInsert(nums, target);

console.log(result);
